src/System.Data.Common/src/System/Data/Common/DbConnectionStringCommon.cs

```diff
@@ -4,86 +4,11 @@
 
 using System.Collections;
 using System.Collections.Generic;
-using System.ComponentModel;
 using System.Diagnostics;
 using System.Globalization;
 
 namespace System.Data.Common
 {
-    internal class DbConnectionStringBuilderDescriptor : PropertyDescriptor
-    {
-        internal DbConnectionStringBuilderDescriptor(string propertyName, Type componentType, Type propertyType, bool isReadOnly, Attribute[] attributes) : base(propertyName, attributes)
-        {
-            ComponentType = componentType;
-            PropertyType = propertyType;
-            IsReadOnly = isReadOnly;
-        }
-
-        internal bool RefreshOnChange { get; set; }
-        public override Type ComponentType { get; }
-        public override bool IsReadOnly { get; }
-        public override Type PropertyType { get; }
-
-        public override bool CanResetValue(object component)
-        {
-            DbConnectionStringBuilder builder = (component as DbConnectionStringBuilder);
-            return ((null != builder) && builder.ShouldSerialize(DisplayName));
-        }
-
-        public override object GetValue(object component)
-        {
-            DbConnectionStringBuilder builder = (component as DbConnectionStringBuilder);
-            if (null != builder)
-            {
-                object value;
-                if (builder.TryGetValue(DisplayName, out value))
-                {
-                    return value;
-                }
-            }
-            return null;
-        }
-
-        public override void ResetValue(object component)
-        {
-            DbConnectionStringBuilder builder = (component as DbConnectionStringBuilder);
-            if (null != builder)
-            {
-                builder.Remove(DisplayName);
-
-                if (RefreshOnChange)
-                {
-                    builder.ClearPropertyDescriptors();
-                }
-            }
-        }
-
-        public override void SetValue(object component, object value)
-        {
-            DbConnectionStringBuilder builder = (component as DbConnectionStringBuilder);
-            if (null != builder)
-            {
-                // via the editor, empty string does a defacto Reset
-                if ((typeof(string) == PropertyType) && string.Empty.Equals(value))
-                {
-                    value = null;
-                }
-                builder[DisplayName] = value;
-
-                if (RefreshOnChange)
-                {
-                    builder.ClearPropertyDescriptors();
-                }
-            }
-        }
-
-        public override bool ShouldSerializeValue(object component)
-        {
-            DbConnectionStringBuilder builder = (component as DbConnectionStringBuilder);
-            return ((null != builder) && builder.ShouldSerialize(DisplayName));
-        }
-    }
-
     [Serializable]
     internal sealed class ReadOnlyCollection<T> : ICollection, ICollection<T>
```

Comment:
@jkotas well, it's needed by https://github.com/dotnet/corefx/pull/18525/files#diff-2572b94a53d7bbf4d82ea0f556f135e0R241 (the only place, other places explicitly use System.Collections.ObjectModel). The only difference between this ROC and the one from System.Collections.ObjectModel - this ROC checks items for [null in DEBUG](https://github.com/dotnet/corefx/blob/master/src/System.Data.Common/src/System/Data/Common/DbConnectionStringCommon.cs#L95-L100), but since https://github.com/dotnet/corefx/blob/master/src/System.Data.Common/src/System/Data/Common/DbConnectionStringBuilder.cs#L238 also does this kind of check - we are safe to remove it - should I?